File: python/grains/grains.py
def square(number):
    if number < 1:
        raise ValueError("Number of squares must be 1 or greater.")
    if number > 64:
        raise ValueError("Number of squares must be less than 64.")
    return 2 ** (number - 1)


def total():
    return 2 ** 64 - 1


if __name__ == '__main__':
    print(f"{square(3)=}")
    print(f"{square(4)=}")
    print(f"{square(5)=}")
    print(f"{square(6)=}")

File: go/grains/grains.go
// Package grains implements a solution for the grains exercism problem
package grains

import (
	"fmt"
)

// Square gives the number of grains on the n-th square on a chess board
func Square(n int) (uint64, error) {
	if !(n >= 1 && n <= 64) {
		return 0, fmt.Errorf("defined only for 1 <= n <= 64, not %v", n)
	}
	return 1 << (n - 1), nil

}

// Total gives the number of grains on the whole chess board
func Total() uint64 {
	return 1<<64 - 1
}
